### ISSUE TITLE
Transparent background behavior breaks several plug-in/extension windows

### ISSUE THREAD

**Issue by kookyabird** [buchizo/ClaudiaIDE#78]
As I understand it the injection of transparency for backgrounds in editors was introduced to address a specific issue, but it has introduced another one. Various editor based windows for other extensions, one being ReSharper, are unusable with a dark theme because the text is too light to be shown on the window background color with the editor being transparent.

An example: The refactoring window in ReSharper has a preset light background. The preview/editing control is an editor which normally has my dark background and light text. ClaudiaIDE attaches itself to the editor control to display my image. When it makes the background transparent, the preset light background of the refactoring window makes it impossible to read the light text.

This occurs in numerous places in ReSharper, and any other extension that uses a non-theme controllable light window background.

I believe a suitable short-term fix would be to add a configuration option to disable the transparency for people like myself who won't encounter the flicker issue, but do encounter this light on light text issue.

**Comment by buchizo** [comment]
@kookyabird Hi, are you talking about transparent image with alpha channel? or ClaudiaIDE's settings? If possible could you provide some case like as screen shots or sample image?

**Comment by kookyabird** [comment]
The image I'm using is a non-transparent PNG. The issue occurs regardless of the opacity or edge fade settings I use. The transparency I was referring to is what you mentioned in this issue here: https://github.com/buchizo/ClaudiaIDE/issues/60#issuecomment-516178890 This behavior you mention has a negative outcome when you have a dark background light text based theme and the window background color for some "editors" is not also dark.

Here are the two example areas without the extension running so that you can see the theming.
![Text Editor - ClaudiaIDE Disabled](https://user-images.githubusercontent.com/31019208/119716791-1a5b6300-be2b-11eb-8f4c-4bbd9e0904b5.png)

![ReSharper Editor Sample - ClaudiaIDE Disabled](https://user-images.githubusercontent.com/31019208/119716792-1a5b6300-be2b-11eb-8b7b-4b5222e392dc.png)

Here are the same areas with the extension running, and no opacity or edge fade settings enabled. Note that the background of the plain text editor has changed to reveal the environment window color as you described in the linked issue comment. The same thing occurs in the ReSharper Options window, wherein the window color is not controllable by the user at all.
![Text Editor - Opacity and Fade](https://user-images.githubusercontent.com/31019208/119716787-19c2cc80-be2b-11eb-9f9a-6dce007c91c6.png)

![ReSharper Editor Sample - Opacity and Fade](https://user-images.githubusercontent.com/31019208/119716790-1a5b6300-be2b-11eb-8979-a13efea35b64.png)

Lastly we have the same areas with my normal settings for opacity and edge fading. Basically the same as the previous, except now I can't even read the text over top of the image because it's 10% opacity, and it's practically the background color.

![Text Editor - No Opacity or Face](https://user-images.githubusercontent.com/31019208/119716783-19c2cc80-be2b-11eb-8c88-3296c72ec72b.png)

![ReSharper Editor Sample - No Opacity or Fade](https://user-images.githubusercontent.com/31019208/119716782-192a3600-be2b-11eb-8df5-731221edc216.png)

This behavior is also seen in the refactoring preview and code generation previews. I don't have it installed anymore, but I had similar behavior in another extension that provided some code stubbing functionality that employed a non-themed window, but used a Visual Studio Editor control for the main work area.

**Comment by buchizo** [comment]
@kookyabird Thank you for your information. I'll think about solves solution for this issue. Maybe I add option that background image limit to main editor window only, if possible.

**Comment by buchizo** [comment]
- new `Background image limit to only editor window` option: false (default)
  - ![image](https://user-images.githubusercontent.com/1254213/119894676-dd50b700-bf77-11eb-9b8a-de9021895498.png)
- new `Background image limit to only editor window` option: true
  - ![image](https://user-images.githubusercontent.com/1254213/119894764-f8bbc200-bf77-11eb-8884-1c4d5a7a54be.png)

**Comment by buchizo** [comment]
I guess this new option is work fine about this issue.

**Comment by buchizo** [comment]
@kookyabird would you try this `Background image limit to only editor window` option after update to 2.2.18?

**Comment by kookyabird** [comment]
It seems the current version available through the VS Marketplace has an issue with the configuration screen. I've uninstalled and reinstalled the extension and I see a simple, "An error occurred loading this property page"

![image](https://user-images.githubusercontent.com/31019208/119898147-0dac3d00-bf07-11eb-8579-f1736283fef9.png)

**Comment by buchizo** [comment]
@kookyabird  sorry, i forgot localized string for English. i quickly fix to 2.2.18.1

**Comment by buchizo** [comment]
I did it.
![image](https://user-images.githubusercontent.com/1254213/119899792-90bcaa00-bf7e-11eb-954a-d8ed10d050bb.png)

**Comment by kookyabird** [comment]
That did it. It's working as your sample did.

I did have an issue where the preview window in ReSharper didn't want to actually render until I:
- Set the new option to false.
- Opened the ReSharper window and selected an item so that the preview rendered.
- Closed the ReSharper windows.
- Set the new option to true.
- Re-opened the ReSharper window and selected an item.

I consider this issue resolved. I'll do some testing to make sure that this weird behavior I just described is a one-off, or if I can make it happen again.

Thank you very much for all the work you've done!
